sections/06_discussion.md

```diff
@@ -25,46 +25,294 @@ human-level performance is irrelevant.*
-@tag:Lanchantin2016_motif @tag:Shrikumar2016_blackbox]. Other studies have
-primarily focused on integrating attention mechanisms with the neural networks.
-Attention mechanisms dynamically weight the importance the neural network gives
-to each feature. By inspecting the attention weights for a particular sample, a
-practitioner can identify the important features for a particular prediction.
-Choi et al [@tag:Choi2016_retain] inverted the typical architecture of recurrent
-neural networks to improve interpretability. In particular, they only used
-recurrent connections in the attention generating procedure, leaving the hidden
-state directly connected to the input variables. In the clinical domain, this
-model was able to produce accurate diagnoses in which the contribution of
-previous hospital visits could be directly interpreted. Choi et al
-[@tag:Choi2016_gram] later extended this work to take into account the structure
-of disease ontologies and found that the concepts represented by the model were
-aligned with medical knowledge. Che et al [@tag:Che2015_distill] introduced a
-knowledge-distillation approach which used gradient boosted trees to learn
-interpretable healthcare features from trained deep models.
+ assumption of normally distributed error. A more recent technique
+ known as test-time dropout [@tag:Gal2015_dropout] can also be
+used to obtain a probabilistic interpretation of a network's outputs.
+
+It can often be informative to understand how the training data
+ affects the learning of a model. Toward this end,
+ Koh & Liang [@tag:Koh2017_understanding] used
+ influence functions, a technique from robust statistics, to
+ trace a model's predictions back through the learning algorithm to identify
+ the datapoints in the training set that had the most impact on a given
+ prediction.
+
+A more free-form approach to interpretability is to visualise
+ the activation patterns of the network on individual inputs and on
+ subsets of the data. ActiVis and CNNvis
+ [@tag:Kahng2017_activis, @tag:Liu2016_towards] are two frameworks that
+ enable interactive
+ visualisation and exploration of large-scale
+ deep learning models.
+
+An orthogonal strategy is to use a knowledge distillation approach
+ to replace a deep learning model with a more interpretable model
+ that achieves comparable performance. Towards this end,
+ Che et al [@tag:Che2015_distill] used gradient boosted trees to learn
+ interpretable healthcare features from trained deep models.
+
+Finally, it is sometimes possible to train the model to
+ provide justifications for its predictions. Lei et al.
+ [@tag:Lei2016_rationalizing] used a generator to identify
+ "rationales", which are short and coherent pieces of the input text that
+ produce similar results to the whole input when passed through an encoder.
+ The authors applied their approach to a sentiment analysis task
+ and obtained substantially superior results compared to an attention-based
+ method.
+
```

Comment:
Are there any final thoughts you would like to add?  This paper is part review, part perspective piece so we have the opportunity to speculate about the future or push certain for certain research directions.

For example:
- Is the criticism of deep learning as a black box, uninterpretable approach exaggerated given all of these methods?
- Are there certain types of features or domains (in genomics, healthcare, etc.) where these methods still fall short?
- Are there practical steps that would improve adoption of appropriate model interpretation?  Maybe closer integration into the most popular deep learning frameworks, as applicable?